src/seedu/addressbook/data/person/Unit.java

```diff
@@ -1,17 +1,34 @@
 package seedu.addressbook.data.person;
 
+import seedu.addressbook.data.exception.IllegalValueException;
+
 public class Unit {
     
     public static final String EXAMPLE = "#02-25";
-    public static final String MESSAGE_BLOCK_CONTRAINTS = "Address unit must be in the format #number-number";
-    public static final String BLOCK_VALIDATION_REGEX = "#[\\d]+-[\\d]+";
+    public static final String MESSAGE_UNIT_CONSTRAINTS = "Address unit must be in the format #number-number";
+    public static final String UNIT_VALIDATION_REGEX = "#[\\d]+-[\\d]+";
     
     private String unitNumber;
     
-    public Unit(String unit) {
+    /**
+     * Validates given address unit.
+     *
+     * @throws IllegalValueException if given address unit string is invalid.
+     */
```

Comment:
Good work in general to write header comments and separating out validations to each class.